x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/ToStringFunctionProcessor.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.eql.expression.function.scalar.string;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xpack.ql.expression.gen.processor.Processor;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class ToStringFunctionProcessor implements Processor {
+
+    public static final String NAME = "sstr";
+
+    private final Processor source;
+
+    public ToStringFunctionProcessor(Processor source) {
+        this.source = source;
+    }
+
+    public ToStringFunctionProcessor(StreamInput in) throws IOException {
+        source = in.readNamedWriteable(Processor.class);
+    }
+
+    @Override
+    public final void writeTo(StreamOutput out) throws IOException {
+        out.writeNamedWriteable(source);
+    }
+
+    @Override
+    public Object process(Object input) {
+        return doProcess(source.process(input));
+    }
+
+    public static Object doProcess(Object source) {
+        return source == null ? "null" : source.toString();
```

Comment:
Why not `null` the instance instead of "null" the string?

Reply:
per https://github.com/elastic/elasticsearch/issues/54465 and https://github.com/endgameinc/eql/blob/master/eql/functions.py#L626 I was following a contract (we can change it, of course) that this function always returns a string. So I catch `null` directly, since I can't call a method on it. Here's the current behavior for how it folds `string(null)`:
```python
>>> import eql
>>> eql.parse_expression("string(null)")
String(value='None')
```

Reply:
Since we're not using some concept of `Optional`, returning `null` instead of a string `"null"` is better since otherwise there's no way to differentiate between a string with `"null"` chars vs actual `null` since they would both be equivalent which is not what we want.